### PR TITLE
Document recursive (c)transpose

### PR DESCRIPTION
**Pull request by andreasnoack**
Rebase/updated version of #15795

### CODE REVIEW

File: base/linalg/rowvector.jl [hunk 1]
```diff
@@ -103,7 +103,8 @@ convert(::Type{RowVector{T,V}}, rowvec::RowVector) where {T,V<:AbstractVector} =
 """
     transpose(v::AbstractVector)
 
-The transposition operator (`.'`).
+The transposition operator (`.'`). Non-recursive behavior may be obtained in such cases
```
Comment (mschauer) [inline base/linalg/rowvector.jl]: Maybe insert `Operates recursively on arrays
 +with vector or matrix element type (e.g., block matrices)` as well here.

**Comment by andyferris** [comment]
Heads up: These changes will class with documentation changes in #24839 (which itself is only waiting on some documentation build stuff I haven't diagonosed (and I'm confused about)).

File: base/linalg/rowvector.jl (hunk 1 above)
Comment (andyferris) [reply base/linalg/rowvector.jl]: Note that it *always* acts recursively, independent of element type. We just have `transpose(x::Number) = x`.

**Comment by andyferris** [comment]
@andreasnoack Other than merging our changes, I think improving these docs is a great idea :)

**Comment by andreasnoack** [comment]
Oh. I didn't realize that #24839 also added documentation to these. If they are overlapping I'll just close this one. Just noticed that #15795 had never been merged and thought it would be easy to finalize.

**Comment by andyferris** [comment]
I didn't mean to imply at all that my documentation was *good* or *better* than this :) There's probably good stuff in both.

**Comment by andreasnoack** [comment]
This is now obsolete with @Sacha0's latest work

**Pull request closed without merging**